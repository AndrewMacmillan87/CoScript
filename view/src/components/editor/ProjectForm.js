import React from "react";

import PropTypes from "prop-types";

class ProjectForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            title: "",
            genres: "",
            description: "",
            cover: [],
        };
        this.onChange = this.onChange.bind(this);
    }

    /**
     * Updates the <[name]> state variable
     * with the text value of the event target
     *
     * @param {Object} e - event object
     */
    onChange(e) {
        const name = e.target.name;
        const value = e.target.value;
        this.setState((state) => ({
            ...state,
            [name]: value,
        }));
    }

    render() {
        const { title, genres, description, cover } = this.state;
        return (
            <aside className='project-form-wrapper'>
                <div className='project-form-container'>
                    <form
                        className='project-form'
                        onSubmit={() =>
                            this.props.onSubmit({
                                title,
                                genres,
                                description,
                                cover,
                            })
                        }>
                        <p className='project-form-heading'>
                            New Project Details
                        </p>
                        <div className='project-form-fields'>
                            <label
                                className='project-form-label'
                                htmlFor='project-title'>
                                Title
                            </label>
                            <input
                                id='project-title'
                                className='project-form-input'
                                name='title'
                                type='text'
                                placeholder='Title...'
                                onChange={(e) => this.onChange(e)}
                                onFocus={(e) => (e.target.placeholder = "")}
                                onBlur={(e) =>
                                    (e.target.placeholder = "Title...")
                                }
                            />
                            <label
                                className='project-form-label'
                                htmlFor='project-genres'>
                                Genres
                            </label>
                            <input
                                id='project-genres'
                                className='project-form-input'
                                name='genres'
                                type='text'
                                placeholder='Comma separated genres...'
                                onChange={(e) => this.onChange(e)}
                                onFocus={(e) => (e.target.placeholder = "")}
                                onBlur={(e) =>
                                    (e.target.placeholder =
                                        "Comma separated genres......")
                                }
                            />
                            <label
                                className='project-form-label'
                                htmlFor='project-description'>
                                Description
                            </label>
                            <textarea
                                id='project-description'
                                className='project-form-textarea'
                                name='description'
                                type='text'
                                placeholder='Description...'
                                onChange={(e) => this.onChange(e)}
                                onFocus={(e) => (e.target.placeholder = "")}
                                onBlur={(e) =>
                                    (e.target.placeholder = "Description...")
                                }></textarea>
                            <div className='project-cover-image-link'>
                                <p>Upload a </p>
                                <label
                                    className='upload-image-text'
                                    htmlFor='project-cover'>
                                    cover image
                                    <input
                                        className='choose-file'
                                        name='cover'
                                        id='project-cover'
                                        type='file'
                                    />
                                </label>
                            </div>
                            <div className='form-submit-btns'>
                                <input
                                    className='file-btn new'
                                    type='submit'
                                    value='Create'></input>
                                <input
                                    className='file-btn cancel'
                                    type='button'
                                    value='Cancel'
                                    onClick={
                                        this.props.removeProjectForm
                                    }></input>
                            </div>
                        </div>
                    </form>
                </div>
            </aside>
        );
    }
}

ProjectForm.propTypes = {
    revealProjectForm: PropTypes.func.isRequired,
    removeProjectForm: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
};

export default ProjectForm;
