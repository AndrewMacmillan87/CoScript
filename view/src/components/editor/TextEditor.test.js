import React, { cloneElement } from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import { TextEditor } from "./TextEditor";
import { FilePanel } from "./FilePanel";
import ProjectForm from "./ProjectForm";
import { ChatBot } from "./ChatBot";

const projectData = require("./test-data/projectData.json");
const fileData = require("./test-data/fileData.json");
const userData = require("./test-data/user.json");

describe("TextEditor component", () => {
    let props,
        mockLoadProject,
        mockGetProjectCollaborators,
        mockLoadFile,
        mockClearFileData,
        mockTextEditor;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        global.HTMLMediaElement.prototype.play = jest.fn();
        mockLoadProject = jest.fn();
        mockGetProjectCollaborators = jest.fn();
        mockLoadFile = jest.fn();
        mockClearFileData = jest.fn();
        mockTextEditor = function () {
            this.textEditor = {
                contentWindow: {
                    document: {
                        designMode: "",
                        body: {
                            innerHTML: "",
                            focus: jest.fn(),
                            addEventListener: jest.fn(),
                            style: {},
                            appendChild: jest.fn(),
                        },
                        addEventListener: jest.fn(),
                    },
                },
            };
        };
        props = {
            user: userData,
            saveDraft: jest.fn(),
            publishProject: jest.fn(),
            loadProject: mockLoadProject,
            getProjectCollaborators: mockGetProjectCollaborators,
            collaboratorsArray: [],
            loadFile: mockLoadFile,
            updateFile: jest.fn(),
            clearFileData: mockClearFileData,
        };
    });

    afterEach(() => {
        props.projectData = null;
        props.fileData = null;
        props.location = null;
    });

    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <TextEditor {...props} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render the FilePanel component on mount", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(FilePanel).length).toEqual(1);
    });

    it("should render the ChatBot component on mount if the projectData prop is a valid object", () => {
        props.projectData = projectData;
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(ChatBot).length).toEqual(1);
    });

    it("should render a main element with a className of text-editor-main", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "text-editor-main" }).length).toEqual(
            1
        );
    });

    it("should render 2 select elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("select").length).toEqual(2);
    });

    it("should render 2 input elements with a className of colourBtn", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "colourBtn" }).length).toEqual(2);
    });

    it("should render a div elements with a className of editor-toolbar", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "editor-toolbar" }).length).toEqual(1);
    });

    it("should render 22 button elements in total", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("button").length).toEqual(22);
    });

    it("should render 19 i elements in total", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("i").length).toEqual(19);
    });

    it("should render an iframe element with a className of text-editor", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("iframe").length).toEqual(1);
        expect(wrapper.find({ className: "text-editor" }).length).toEqual(1);
    });

    it("Should call `loadFile()` action creator if there is some file data", () => {
        props.fileData = fileData;
        props.location = { pathname: null };
        jest.spyOn(TextEditor.prototype, "setTextEditor").mockImplementation(
            mockTextEditor
        );
        mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        expect(mockLoadFile).toBeCalled();
    });

    it("Should call `loadProject()` and `clearFileData()` action creators if there is some project data", async () => {
        props.projectData = projectData;
        props.location = { pathname: null };
        jest.spyOn(TextEditor.prototype, "setTextEditor").mockImplementation(
            mockTextEditor
        );
        mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        await expect(mockLoadProject).toBeCalled();
        expect(mockClearFileData).toBeCalled();
    });

    it("Should call `getProjectCollaborators()` action creator when component updates", async () => {
        props.projectData = projectData;
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        await wrapper.update();
        expect(mockGetProjectCollaborators).toBeCalled();
    });
});

describe("FilePanel component", () => {
    let textEditorProps,
        props,
        mockPublishProject,
        mockGetAllProjects,
        mockCreateNewFolder,
        mockCreateNewFile;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        mockPublishProject = jest.fn();
        mockGetAllProjects = jest.fn();
        mockCreateNewFolder = jest.fn();
        mockCreateNewFile = jest.fn();
        textEditorProps = {
            user: userData,
            saveDraft: jest.fn(),
            publishProject: mockPublishProject,
            loadProject: jest.fn(),
            getProjectCollaborators: jest.fn(),
            collaboratorsArray: [],
            loadFile: jest.fn(),
            updateFile: jest.fn(),
            clearFileData: jest.fn(),
        };
        props = {
            revealFilePanel: true,
            revealProjectForm: jest.fn(),
            revealPublishWindow: jest.fn(),
            cancelPublish: jest.fn(),
            submitPublish: jest.fn(),
            projectForm: false,
            collaborators: [],
            revealPublish: false,
            onClick: jest.fn(),
            onSubmit: jest.fn(),
            loadFile: jest.fn(),
            loadProject: jest.fn(),
            clearFileData: jest.fn(),
            clearProjectData: jest.fn(),
            getAllProjects: mockGetAllProjects,
            createNewFolder: mockCreateNewFolder,
            createNewFile: mockCreateNewFile,
        };
    });

    afterEach(() => {
        props.projectData = null;
    });

    it("should render an aside element with a className of file-panel-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <FilePanel {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("aside").length).toEqual(1);
        expect(
            wrapper.find({ className: "file-panel-container" }).length
        ).toEqual(1);
    });

    it("should render with with 5 buttons on mount", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <FilePanel {...props} />
                </Router>
            </Provider>
        );
        const filePanelWrapper = wrapper.find(FilePanel);
        expect(filePanelWrapper.find("button").length).toEqual(5);
    });

    it("should change the 'file-panel' className to 'file-panel reveal' when the reveal button is clicked", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        expect(
            filePanelWrapper.find({ className: "file-panel reveal" }).length
        ).toEqual(1);
    });

    it("should render the ProjectForm component when the 'Create' button is clicked", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find(ProjectForm).length).toEqual(1);
    });

    it("should render the publish form when the 'Publish' button is clicked", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn publish" })
            .at(0)
            .simulate("click");

        expect(
            wrapper.find({ className: "publish-project-container" }).length
        ).toEqual(1);
    });

    it("should render the new folder form when the 'Folder' button is clicked", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn" })
            .at(0)
            .simulate("click");

        expect(
            wrapper.find({ className: "new-folder-container" }).length
        ).toEqual(1);
    });

    it("should render the new file form when the 'File' button is clicked", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn" })
            .at(1)
            .simulate("click");

        expect(
            wrapper.find({ className: "new-folder-container" }).length
        ).toEqual(1);
    });

    it("should call publishProject when the publish form is submitted", async () => {
        textEditorProps.user = { userId: projectData.user };
        textEditorProps.projectData = projectData;
        textEditorProps.history = { push: jest.fn() };
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );

        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");
        wrapper.find({ className: "file-btn publish" }).at(0).simulate("click");

        await act(async () => {
            wrapper
                .find({ className: "project-publish-form" })
                .simulate("submit", { preventDefault: jest.fn() });
        });
        expect(mockPublishProject).toBeCalled();
    });
});

describe("ProjectForm component", () => {
    let props;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        props = {
            user: userData,
            saveDraft: jest.fn(),
            publishProject: jest.fn(),
            loadProject: jest.fn(),
            getProjectCollaborators: jest.fn(),
            collaboratorsArray: [],
            loadFile: jest.fn(),
            updateFile: jest.fn(),
            clearFileData: jest.fn(),
        };
    });

    it("should render an aside element (2 in total) with a className of project-form-wrapper", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find("aside").length).toEqual(2);
        expect(
            wrapper.find({ className: "project-form-wrapper" }).length
        ).toEqual(1);
    });

    it("should render a form element with a className of project-form-wrapper", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find("form").length).toEqual(1);
        expect(wrapper.find({ className: "project-form" }).length).toEqual(1);
    });

    it("should render 4 label elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find("label").length).toEqual(4);
    });

    it("should render 5 input elements (7 in total)", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find("input").length).toEqual(7);
    });

    it("should render 1 textarea element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...props} />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        expect(wrapper.find("textarea").length).toEqual(1);
    });

    it("should call history.push when the project form is submitted", async () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor
                        {...props}
                        projectData={projectData}
                        history={{ push: jest.fn() }}
                    />
                </Router>
            </Provider>
        );
        wrapper.find({ className: "panel-btn" }).at(0).simulate("click");

        const filePanelWrapper = wrapper.find(FilePanel);
        filePanelWrapper
            .find({ className: "file-btn new" })
            .at(0)
            .simulate("click");

        const fakeEvent = {
            preventDefault: () => {},
        };

        wrapper
            .find({ className: "project-form" })
            .simulate("submit", fakeEvent);

        expect(
            wrapper.props().children.props.children.props.history.push
        ).toBeCalled();
    });
});

describe("ChatBot component", () => {
    let textEditorProps, props, mockGetMessages, mockSendMessage;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        mockGetMessages = jest.fn();
        mockSendMessage = jest.fn();
        textEditorProps = {
            user: userData,
            saveDraft: jest.fn(),
            publishProject: jest.fn(),
            loadProject: jest.fn(),
            getProjectCollaborators: jest.fn(),
            collaboratorsArray: [],
            loadFile: jest.fn(),
            updateFile: jest.fn(),
            clearFileData: jest.fn(),
            projectData: projectData,
        };
        props = {
            user: {},
            projectData: projectData,
            revealChatBot: true,
            onClick: jest.fn(),
            getMessages: mockGetMessages,
            sendMessage: mockSendMessage,
        };
    });

    it("should call getMessages on update", async () => {
        jest.spyOn(ChatBot.prototype, "componentDidUpdate").mockImplementation(
            mockGetMessages()
        );
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ChatBot {...props} />
                </Router>
            </Provider>
        );
        wrapper.setProps({});
        await expect(mockGetMessages).toBeCalled();
    });

    it("should render an aside element with a className of chat-bot-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );

        const chatWrapper = wrapper.find(ChatBot);
        expect(chatWrapper.find("aside").length).toEqual(1);
        expect(
            chatWrapper.find({ className: "chat-bot-container" }).length
        ).toEqual(1);
    });

    it("Should render a div elements with className of chat-window", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "chat-window" }).length).toEqual(1);
    });

    it("Should render a form elements with className of editor-chat-form", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("form").length).toEqual(1);
        expect(wrapper.find({ className: "editor-chat-form" }).length).toEqual(
            1
        );
    });

    it("Should render 2 button elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        const chatWrapper = wrapper.find(ChatBot);
        expect(chatWrapper.find("button").length).toEqual(2);
    });

    it("Should render 2 i elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );

        const chatWrapper = wrapper.find(ChatBot);
        expect(chatWrapper.find("i").length).toEqual(2);
    });

    it("should call sendMessage when the chat form is submitted", async () => {
        jest.spyOn(ChatBot.prototype, "onClickSend").mockImplementation(
            mockSendMessage()
        );
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <TextEditor {...textEditorProps} />
                </Router>
            </Provider>
        );
        await expect(mockSendMessage).toBeCalled();
    });
});
