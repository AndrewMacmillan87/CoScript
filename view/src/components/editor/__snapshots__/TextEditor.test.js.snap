// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor component renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  />,
  <aside
    className="file-panel-container"
  >
    <div
      className="file-panel"
    >
      <div
        className="file-menu-container"
      >
        <button
          className="file-btn new"
          onClick={[Function]}
          type="button"
        >
          New
        </button>
        <button
          className="file-btn publish"
          onClick={[Function]}
          type="button"
        >
          Publish
        </button>
        <button
          className="file-btn"
          onClick={[Function]}
          type="button"
        >
          Folder
        </button>
        <button
          className="file-btn"
          onClick={[Function]}
          type="button"
        >
          File
        </button>
      </div>
      <div
        className="panel-title-container"
      >
        <h3
          className="panel-title panel-text"
        >
          My Files
        </h3>
      </div>
      <div
        className="files"
      >
        <div
          className="project-folders-container"
        />
      </div>
      <div
        className="panel-title-container"
      >
        <h3
          className="panel-title panel-text"
        >
          Friends Online
        </h3>
      </div>
    </div>
    <button
      className="panel-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="reveal-arrow"
      >
        â€º
      </span>
    </button>
  </aside>,
  <main
    className="text-editor-main"
  >
    <div
      className="text-editor-ui-container"
    >
      <div
        className="editor-toolbar"
      >
        <div
          className="font-menus"
        >
          <select
            className="font-menu"
            onChange={[Function]}
          >
            <option
              value="Times New Roman"
            >
              Times New Roman
            </option>
            <option
              value="Consolas"
            >
              Consolas
            </option>
            <option
              value="Tahoma"
            >
              Tahoma
            </option>
            <option
              value="Calibri"
            >
              Calibri
            </option>
            <option
              value="Arial"
            >
              Arial
            </option>
            <option
              value="Verdana"
            >
              Verdana
            </option>
            <option
              value="Trebuchet MS"
            >
              Trebuchet MS
            </option>
            <option
              value="Courier New"
            >
              Courier New
            </option>
            <option
              value="Comic Sans MS"
            >
              Comic Sans MS
            </option>
            <option
              value="Brush Script MT"
            >
              Brush Script MT
            </option>
            <option
              value="Impact"
            >
              Impact
            </option>
          </select>
          <select
            className="font-menu"
            onChange={[Function]}
          >
            <option
              value={0}
            >
              8
            </option>
            <option
              value={1}
            >
              9
            </option>
            <option
              value={2}
            >
              10
            </option>
            <option
              value={3}
            >
              11
            </option>
            <option
              value={4}
            >
              12
            </option>
            <option
              value={5}
            >
              13
            </option>
            <option
              value={6}
            >
              14
            </option>
            <option
              value={7}
            >
              15
            </option>
          </select>
        </div>
        <i
          className="fas fa-palette colour-icon"
        />
        <input
          className="colourBtn"
          id="font-colour"
          onChange={[Function]}
          title="Change Font Colour"
          type="color"
        />
        <i
          className="fas fa-highlighter colour-icon"
        />
        <input
          className="colourBtn"
          onChange={[Function]}
          title="Highlight"
          type="color"
        />
        <button
          className="boldBtn"
          onClick={[Function]}
          title="Bold"
        >
          <i
            className="fas fa-bold"
          />
        </button>
        <button
          className="italicBtn"
          onClick={[Function]}
          title="Italic"
        >
          <i
            className="fas fa-italic"
          />
        </button>
        <button
          className="underlineBtn"
          onClick={[Function]}
          title="Underlined"
        >
          <i
            className="fas fa-underline"
          />
        </button>
        <button
          className="subscriptBtn"
          onClick={[Function]}
          title="Subscript"
        >
          <i
            className="fas fa-subscript"
          />
        </button>
        <button
          className="superScriptBtn"
          onClick={[Function]}
          title="Superscript"
        >
          <i
            className="fas fa-superscript"
          />
        </button>
        <button
          className="strikeBtn"
          onClick={[Function]}
          title="Strike-through"
        >
          <i
            className="fas fa-strikethrough"
          />
        </button>
        <button
          className="rmvFormatBtn"
          onClick={[Function]}
          title="Remove Formatting"
        >
          <i
            className="fas fa-remove-format"
          />
        </button>
        <button
          className="ulBtn"
          onClick={[Function]}
          title="Bullet Points"
        >
          <i
            className="fas fa-list-ul"
          />
        </button>
        <button
          className="oltBtn"
          onClick={[Function]}
          title="Numbered List"
        >
          <i
            className="fas fa-list-ol"
          />
        </button>
        <button
          className="leftAlignBtn"
          onClick={[Function]}
          title="Align Left"
        >
          <i
            className="fas fa-align-left"
          />
        </button>
        <button
          className="centerAlignBtn"
          onClick={[Function]}
          title="Align Center"
        >
          <i
            className="fas fa-align-center"
          />
        </button>
        <button
          className="rightAlignBtn"
          onClick={[Function]}
          title="Align Right"
        >
          <i
            className="fas fa-align-right"
          />
        </button>
        <button
          className="justifyAlignBtn"
          onClick={[Function]}
          title="Align Justify"
        >
          <i
            className="fas fa-align-justify"
          />
        </button>
        <button
          className="undoBtn"
          onClick={[Function]}
          title="Undo"
        >
          <i
            className="fas fa-undo"
          />
        </button>
        <button
          className="redoBtn"
          onClick={[Function]}
          title="Redo"
        >
          <i
            className="fas fa-redo"
          />
        </button>
        <button
          className="linkBtn"
          onClick={[Function]}
          title="Link"
        >
          <i
            className="fas fa-link"
          />
        </button>
        <button
          className="unlinkBtn"
          onClick={[Function]}
          title="Unlink"
        >
          <i
            className="fas fa-unlink"
          />
        </button>
      </div>
    </div>
    <div
      className="text-editor-wrapper"
    >
      <iframe
        className="text-editor"
        frameBorder="0"
        id="text-editor"
        name="text-editor"
        title="Rich editor"
      />
    </div>
  </main>,
]
`;
